jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialManager.kt

```diff
@@ -146,14 +149,28 @@ class DefaultCredentialManager : CredentialManager() {
     }
 }
 
+private val CREDENTIAL_ENVIRONMENT_VARIABLES = setOf(
+    AWS_ACCESS_KEY_ID.environmentVariable(),
+    AWS_SECRET_ACCESS_KEY.environmentVariable(),
+    AWS_SESSION_TOKEN.environmentVariable()
+)
+
 fun AwsCredentials.toEnvironmentVariables(): Map<String, String> {
     val map = mutableMapOf<String, String>()
-    map["AWS_ACCESS_KEY_ID"] = this.accessKeyId()
-    map["AWS_SECRET_ACCESS_KEY"] = this.secretAccessKey()
+    map[AWS_ACCESS_KEY_ID.environmentVariable()] = this.accessKeyId()
+    map[AWS_SECRET_ACCESS_KEY.environmentVariable()] = this.secretAccessKey()
 
     if (this is AwsSessionCredentials) {
-        map["AWS_SESSION_TOKEN"] = this.sessionToken()
+        map[AWS_SESSION_TOKEN.environmentVariable()] = this.sessionToken()
     }
 
     return map
 }
+
+fun AwsCredentials.toEnvironmentVariables(existing: MutableMap<String, String>, replace: Boolean = false) {
+    val envVars = toEnvironmentVariables()
+    if (replace || existing.keys.none { it in CREDENTIAL_ENVIRONMENT_VARIABLES }) {
```

Comment:
This will fail if you replace session with static access/secret

Reply:
You're worried about the `AWS_SESSION_TOKEN` being left dangling? 
That's why we do `CREDENTIAL_ENVIRONMENT_VARIABLES.forEach { existing.remove(it) }` to make sure that all existing "credential environment variables" get removed from the map.
There's also a test around it see: https://github.com/aws/aws-toolkit-jetbrains/pull/2151/files#diff-d0ef7b3e36ea585764aa7da10d6501228077894277b9a200d9e7529d29afbcbdR53

Reply:
Welp I'm blind